svgdigitizer/electrochemistry/cv.py

```diff
@@ -18,111 +18,166 @@
 #
 #  You should have received a copy of the GNU General Public License
 #  along with svgdigitizer. If not, see <https://www.gnu.org/licenses/>.
-#*********************************************************************
+# ********************************************************************
+from functools import cache
 import pandas as pd
 import matplotlib.pyplot as plt
 from pathlib import Path
+from astropy import units as u
+
+# The keys in this dict are unit strings that are identical to those
+# obtained by using str() on an astropy unit of a quantity, i.e.,
+# >>> from astropy import units as u
+# >>> q = 10 * u.A / u.cm**2
+# >>> str(q.unit)
+# 'A / cm2'
+# These string can directly be converted to a unit
+# >>> u.Unit('uA / cm2')
+unit_typos = {'uA / cm2': ['uA / cm2', 'uA / cm²', 'µA / cm²', 'µA cm⁻²', 'uA cm-2', 'uA / cm2'],
+              'A / cm2': ['A / cm2', 'A cm⁻²', 'A cm-2', 'A / cm2'],
+              'A': ['A', 'ampere', 'amps', 'amp'],
+              'mV': ['milliV', 'millivolt', 'milivolt', 'miliv', 'mV'],
+              'V': ['V', 'v', 'Volt', 'volt'],
+              'V / s': ['V s-1', 'V / s'],
+              'mV / s': ['mV / s', 'mV s-1', 'mV/s']}
 
 
 class CV():
-    def __init__(self, metadata, svgplot):
+    def __init__(self, svgplot, metadata=None):
         """
         metadata: dict
         """
         self.svgplot = svgplot
-        self.metadata = metadata
-        self.svgplot.create_df()
+        self.metadata = metadata or {}
 
-        # TODO: These labels should either be extracted from the svg or yaml file
-        self.xlabel = 'U'
+    @property
+    @cache
+    def axis_properties(self):
+        return {'x': {'dimension': 'U',
+                      'unit': 'V'},
+                'y': {'dimension': 'j' if 'm2' in str(self.get_axis_unit('y')) else 'I',
+                      'unit': 'A / m2' if 'm2' in str(self.get_axis_unit('y')) else 'A'}}
 
-        # TODO: should be j if both current and normalized to are given in the yaml file
-        self.ylabel = 'I'
+    def get_axis_unit(self, axis):
+        r'''
+        Replaces the units derived from the svg file into astropy units.
+        '''
+        unit = self.svgplot.units[axis]
 
-        # TODO: All the rest in the init is presumably not necessary
-        self.description = self.metadata['figure description']
+        return self.get_correct_unit(unit)
 
-        self.xunit = self.description['potential scale']['unit']
+    @classmethod
+    def get_correct_unit(cls, unit):
+        for correct_unit, typos in unit_typos.items():
+            if unit in typos:
+                return u.Unit(correct_unit)
 
-        self.yunit = self.description['current']['unit']
+        raise ValueError(f'Unknown Unit {unit} on Axis {axis}')
 
-        self.get_rate()
+    @property
+    @cache
+    def rate(self):
+        r'''
+        Return a rate based on a label in the SVG file.
```

Comment:
Could you elaborate in the docstring what a `rate` is?

It's also usually good to split such comments into "what does it do" and "how does it do it" such as:
```
Return the scan rate of the plot.

The scan rate is read from a `<text>` in the SVG file such as `<text>scan rate: 1abc</text>`.
```